go/cmd/dolt/commands/init.go

```diff
@@ -151,7 +153,8 @@ func (cmd InitCmd) Exec(ctx context.Context, commandStr string, args []string, d
 		}
 	}
 
-	err := dEnv.InitRepoWithTime(context.Background(), types.Format_Default, name, email, initBranch, t)
+	requiresFunHash := apr.Contains(requiresFunHashName)
+	err := dEnv.InitRepoWithTime(context.Background(), types.Format_Default, name, email, initBranch, t, requiresFunHash)
```

Comment:
Trouble is, it's not just `InitRepoWithTime`.  `InitRepoWithTime` calls `InitDBAndRepoState` which calls `InitDBWithTime` which calls `WriteEmptyRepoWithCommitTime` which calls `WriteEmptyRepoWithCommitTimeAndDefaultBranch`. Each of these methods is visible outside the package and many of them have multiple callers that needed to be updated, and some of them contain logic that, if we make new functions, will either need to be duplicated or outlined.

I don't think there's going to be a tidy solution to this sadly (at least in a language without optional parameters). Having the extra argument and updating the callsites is not great, but I think trying to make new functions for all of them is worse because of how it would complicate/duplicate the logic in them.

Fortunately, it doesn't look like any of these functions are called outside of the doltcore library, so other libraries don't need to worry about this in any case.

Reply:
Ahh... yes... I followed this a little closer and see what you mean. It would require forking a bunch of calls and I think you're right that it wouldn't necessarily end up making things simpler. Thank you for taking the time to think about the idea!